bin/update-ts-project-references.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+import cp from "child_process";
+import fs from "fs";
+import fsP from "fs/promises";
+import path from "path";
+import { getDependentsGraph } from "@changesets/get-dependents-graph";
+import { getPackages } from "@manypkg/get-packages";
+
+type GetPackagesResult = Awaited<ReturnType<typeof getPackages>>;
+type Package = GetPackagesResult["packages"][number];
+type DependencyGraph = Map<string, Set<string>>;
+type TSConfig = { references?: { path: string }[] };
+
+const stringArrayEquals = (arrA: string[], arrB: string[]): boolean => {
+  return (
+    arrA.length === arrB.length && arrA.every((itemA, i) => itemA === arrB[i])
+  );
+};
+
+let prettier: typeof import("prettier") | undefined;
```

Comment:
shouldn't we just make sure prettier is always installed as a requirement? its very unlikely that it isn't

I mean if I understand correctly this code is trying to make it optional

Reply:
This is not about making it optional - it will still throw when updating the file if Prettier is missing. It's just lazy-loading it to make this script faster when there are no updates to be made (which should be true for most of its runs). I will add a comment explaining this

Reply:
got it, maybe worth the comment, wasn't obvious